code/kobe.py

```diff
@@ -1107,32 +1117,543 @@ def kobe_shadows(df_input,cdpp):
+        print('----------------------------------')
+    
+    return df_kobe_output
+
+
 def kobe_transits(df_kobe_output):
     """
     The KOBE Transits module: calculate transit related parameters for all planets
 
     Parameters
     ----------
-    df_input : pandas dataframe
-        input dataframe processed suitably
-
+    df_kobe_output : pandas dataframe
+        dataframe of planets that transit (potentially detectable) - input dataframe processed suitably
+        columns: all from input, some are added
+    
+    Output 
+    ------
+    df_kobe_output : pandas dataframe
+        dataframe of all KOBE Shadows planets with transit-related parameters are calculated
+        columns: all from input, some are added
+    
     """
 
+    # Assigning required values to variables
+    # Argument of periastron
+    arg_periastron = np.pi/2 - df_kobe_output['kobe_observer_azimuth']
+    
     # Calculate impact parameter for each planet with respect to the observer
     # we use observer's azimuth as true anomaly --> implies that planet is crossing observer's azimuth
     # we use observer's polar angle as inclination --> identical by definition
     distance_to_star = (df_kobe_output[col_sma]*sp_constants.astronomical_unit*(1-(df_kobe_output[col_ecc]**2)))/(1 + (df_kobe_output[col_ecc]*np.cos(df_kobe_output['kobe_observer_azimuth'])))
     df_kobe_output['impact_parameter'] = (distance_to_star * np.cos(df_kobe_output['kobe_inclination']))/(df_kobe_output[col_r_star]*radius_sun.value)
-
+    
+    # Include the effect of eccentricity and calculate impact parameter accordingly 
+    if include_ecc == True:
+        # Defining an extra variable 'rho_c' as per David M. Kipping, 2010
+        # eccentric_factor = (np.sqrt(1-((df_kobe_output[col_ecc])**2))/(1+(df_kobe_output[col_ecc]*np.sin(arg_periastron))))
+        rho_c = (1-(df_kobe_output[col_ecc]**2))/(1+(df_kobe_output[col_ecc]*np.sin(arg_periastron)))
+        ar = (df_kobe_output[col_sma]*sp_constants.astronomical_unit)/(df_kobe_output[col_r_star]*radius_sun.value)
+        df_kobe_output['impact_parameter'] = (ar)*rho_c*np.cos(df_kobe_output['kobe_inclination'])
+        
+    
     # geometrically, transits occur if impact parameter (sky projected distance of planet from star in stellar radius unit) is less than the sum of planet and stellar radii in stellar radius unit
-    # we calculate the this geometric qunatity
+    # we calculate this geometric quantity
     df_kobe_output['rp+rs/rs'] = 1 + ((df_kobe_output[col_r_planet]/df_kobe_output[col_r_star])*(radius_earth/radius_sun))
 
+    # KOBE 1.0
+    # characteristic transit duration : eq 19, Transits and Occultations by J. Winn in Exoplanets
+    df_kobe_output['transit_duration [hours]'] = ((df_kobe_output[col_r_star]*radius_sun.value)*(df_kobe_output[col_period]*24))/((np.pi)*(df_kobe_output[col_sma]*sp_constants.astronomical_unit))
```

Comment:
right now, both kobe1 and kobe2 calculations are occurring. this seems unnecessary?